File: servico-musica-java/java.soap/src/main/java/servico/java/soap/entity/Usuario.java
package servico.java.soap.entity;

import jakarta.persistence.*;

import java.util.ArrayList;
import java.util.List;

@Entity
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String nome;
    private short idade;

    @OneToMany(mappedBy = "usuario",cascade = CascadeType.ALL)
    private List<Playlist> playlistsUsuario = new ArrayList<>();

    protected Usuario(){}

    public Usuario(String nome, short idade) {
        this.nome = nome;
        this.idade = idade;
    }

    public long getId() {
        return id;
    }

    public String getNome() {
        return nome;
    }

    public short getIdade() {
        return idade;
    }

    public List<Playlist> getPlaylistsUsuario() {
        return playlistsUsuario;
    }
}

File: servico-musica-kotlin/kotlin.soap/src/main/kotlin/servico/kotlin/soap/entity/Usuario.kt
package servico.kotlin.soap.entity

import jakarta.persistence.*

@Entity
class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    var nome: String? = null

    var idade: Short = 0

    @OneToMany(mappedBy = "usuario", cascade = [CascadeType.ALL])
    var playlistsUsuario: MutableList<Playlist> = mutableListOf()

    constructor()

    constructor(nome: String?, idade: Short) {
        this.nome = nome
        this.idade = idade
    }
}